web-server/v0.4/mock/api.js

```diff
@@ -1,64 +1,60 @@
 import config from './datastoreConfig';
 
-export const mockControllerAggregation = {
-  aggregations: {
-    controllers: {
-      buckets: [
-        {
-          key: 'a_test_controller',
-          doc_count: 1,
-          runs_preV1: { value: null },
-          runs: { value: 1, value_as_string: '1111-11-11' },
-        },
-        {
-          key: 'b_test_controller',
-          doc_count: 2,
-          runs_preV1: { value: null },
-          runs: { value: 2, value_as_string: '2222-22-22' },
-        },
-      ],
+// Generates mock controllers
+export const generateMockControllerAggregation = () => {
+  const mockControllerSchema = {
+    aggregations: {
+      controllers: {
+        buckets: [],
+      },
     },
-  },
+  };
+  for (let i = 0; i <= 25; i += 1) {
+    mockControllerSchema.aggregations.controllers.buckets.push({
+      key: `${(i + 10).toString(36)}_test_controller`,
+      doc_count: i,
+      runs_preV1: { value: null },
+      runs: {
+        value: `${i + 1}`,
+        value_as_string: `${i + 1}${i + 1}${i + 1}${i + 1}-${i + 1}${i + 1}-${i + 1}${i + 1}`,
+      },
+    });
+  }
+  return mockControllerSchema;
 };
 
 const datastoreConfig = config['/dev/datastoreConfig'];
 const prefix = datastoreConfig.prefix + datastoreConfig.run_index.slice(0, -1);
-export const mockIndices = [
-  {
-    index: `${prefix}.0000-00-00`,
-  },
-  {
-    index: `${prefix}.0000-00-01`,
-  },
-];
+// Generates mock indices
+export const generateMockIndices = () => {
+  const indexSchema = [];
+  for (let i = 0; i <= 25; i += 1) {
+    indexSchema.push({ index: i < 10 ? `${prefix}.0000-00-0${i}` : `${prefix}.0000-00-${i}` });
+  }
+  return indexSchema;
+};
 
-export const mockResults = {
-  hits: {
-    hits: [
-      {
-        fields: {
-          'run.name': ['a_test_run'],
-          '@metadata.controller_dir': ['test_run.test_domain.com'],
-          'run.start': ['1111-11-11T11:11:11+00:00'],
-          'run.id': ['1111'],
-          'run.end': ['1111-11-11T11:11:12+00:00'],
-          'run.controller': ['test_run.test_domain.com'],
-          'run.config': ['test_size_1'],
-        },
-      },
-      {
-        fields: {
-          'run.name': ['b_test_run'],
-          '@metadata.controller_dir': ['b_test_run.test_domain.com'],
-          'run.start': ['1111-11-11T11:11:13+00:00'],
-          'run.id': ['2222'],
-          'run.end': ['1111-11-11T11:11:14+00:00'],
-          'run.controller': ['b_test_run.test_domain.com'],
-          'run.config': ['test_size_2'],
-        },
+// Generates mock results
+export const generateMockResults = () => {
+  const mockResultsSchema = {
+    hits: {
+      hits: [],
+    },
+  };
+  for (let i = 0; i <= 25; i += 1) {
```

Comment:
Running from 0 to 25 inclusively gives us 26 results.  It helps to add a comment that that is intentional.  Alternatively, you can have the loop start at 1 and go to 26, `for (let i = 1; i <= 26; i += 1)` or from zero, `for (let i = 0; i < 26; i += 1)`, either way the `26` is in the loop to make it easier to understand 26 items are expected.

If we agree to have for options as described below, we might want to have the mock instance derive the number of items to generate from the code itself instead of hard-coding the value of `26`.

Reply:
This comment applies to the loop at line 12 as well.